tests/utils/test_encrypted_file_metadata.py

```diff
@@ -0,0 +1,92 @@
+#  Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import copy
+import unittest
+
+from matrix_content_scanner.utils.constants import ErrCode
+from matrix_content_scanner.utils.encrypted_file_metadata import (
+    validate_encrypted_file_metadata,
+)
+from matrix_content_scanner.utils.errors import ContentScannerRestError
+from tests.testutils import ENCRYPTED_FILE_METADATA
+
+
+class EncryptedMetadataValidationTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        self.metadata = copy.deepcopy(ENCRYPTED_FILE_METADATA)
+
+    def test_validate(self) -> None:
+        """Tests that valid file metadata is considered as such."""
+        validate_encrypted_file_metadata(ENCRYPTED_FILE_METADATA)
+
+    def test_key_ops_no_decrypt(self) -> None:
+        """Tests that the metadata validation fails if key_ops doesn't include `decrypt`."""
+        self.metadata["file"]["key"]["key_ops"] = ["encrypt"]
+        self._test_fails_validation()
+
+    def test_key_ops_no_encrypt(self) -> None:
+        """Tests that the metadata validation fails if key_ops doesn't include `encrypt`."""
+        self.metadata["file"]["key"]["key_ops"] = ["decrypt"]
+        self._test_fails_validation()
+
+    def test_no_file(self) -> None:
+        """Tests that the metadata validation fails if there isn't a `file` property."""
+        self.metadata = {"foo": "bar"}
+        self._test_fails_validation()
+
+    def test_no_key(self) -> None:
+        """Tests that the metadata validation fails if there isn't a `file.key` property."""
+        del self.metadata["file"]["key"]
+        self._test_fails_validation()
+
+    def test_no_k(self) -> None:
```

Comment:
Is it worth having tests for these being the incorrect values?

Reply:
Do you mean a key that cannot decrypt the file? Or a value of incorrect type?

If the former we already test it here:

https://github.com/matrix-org/matrix-content-scanner-python-wip/blob/7c06deee8653c7bfea44ccb633340130c74ed659/tests/scanner/test_scanner.py#L190-L214

Reply:
I put this comment on somewhat a random choice and...chose poorly! 😢 

I mean for the fields that are constant values, should we add tests that they are rejected if they have the *wrong* value, not just a missing value.

Reply:
Oh I see. Didn't help that `k` is the one field in `file.key` that isn't a constant value. But yeah, more testing always is a good idea.

Reply:
Done in https://github.com/matrix-org/matrix-content-scanner-python/pull/25/commits/1876b0c54f65ac3f70f57dcdb99dc83b06f1f662